p2p/pex/pex_reactor.go

```diff
@@ -517,8 +525,7 @@ func (r *PEXReactor) dialAttemptsInfo(addr *p2p.NetAddress) (attempts int, lastD
 
 func (r *PEXReactor) dialPeer(addr *p2p.NetAddress) error {
 	attempts, lastDialed := r.dialAttemptsInfo(addr)
-
-	if attempts > maxAttemptsToDial {
+	if !r.Switch.IsPeerPersistent(addr) && attempts > maxAttemptsToDial {
```

Comment:
**note**: this changes previously defined behavior where persistent peer could've been marked as BAD

Reply:
yes. it is correct. now persistent peer is so important that we dial them even if it is not available for very long time.(of course, behavior is controllable via `PersistentPeersMaxDialPeriod`)

if we want to separate the definition of "persistent peer"(indefinite dialing) and "peer"(no more dialing after some period), then we need another parameter in config.

because I think dialing several persistent peers every n minutes is not a big burden for the state machine, indefinite dialing for all persistent peers practically make sense to me.

how do you think @melekes ?

Reply:
I think it's fine to continue to dial persistent peers indefinitely. We just need to document that change in the specification https://github.com/tendermint/spec/blob/master/spec/reactors/pex/pex.md (this part: "If we fail to connect to the peer after 16 tries (with exponential backoff), we remove from address book completely.")

Reply:
We just updated it here https://github.com/tendermint/spec/pull/68